rclc/CMakeLists.txt

```diff
@@ -1,56 +1,126 @@
 cmake_minimum_required(VERSION 3.5)
+set(CMAKE_C_CLANG_TIDY clang-tidy -checks=*)
 
 project(rclc)
 
+#################################################
+# compiler settings
+#################################################
+
+# Default to C11
+if(NOT CMAKE_C_STANDARD)
+  set(CMAKE_C_STANDARD 11)
+endif()
+
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
```

Comment:
This package is a C-only package and doesn't require any C++ compiler, does it? If so, I don't think these parameters make much of an impact.

Reply:
The unit tests are written in C++.